### PR TITLE
Team4 MVP

### PR DESCRIPTION
**Pull request by hussain2603**
# Summary

An Evolutionary agent that tries to optimise the food eaten. 

The python, json, and bash files will not be used during the actual run. Only for learning purposes for our agent.

### CODE REVIEW

File: pkg/agents/team4/agent1/agent.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+func (a *CustomAgentEvo) Run() {
+
+	// Using custom params to decide how much food to eat
+	scaledFloorScore := a.params.currentFloorScore.EvaluateEquation(a.Floor()) / a.params.scalingEquation.EvaluateEquation(a.Floor())
+	scaledHpScore := a.params.currentHpScore.EvaluateEquation(a.HP()) / a.params.scalingEquation.EvaluateEquation(a.HP())
+	foodToEat := math.Max(0.0, 50*scaledFloorScore+50*scaledHpScore)
+
+	beforeHP := a.HP()
+	a.TakeFood(food.FoodType(foodToEat))
+	foodEaten := a.HP() - beforeHP
```
Comment (jzzheng22) [inline pkg/agents/team4/agent1/agent.go:97]:
```suggestion
	foodEaten := a.TakeFood(foodToEat)
```

File: pkg/agents/team4/agent1/agent.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+		},
+	}, nil
+}
+
+func (a *CustomAgentEvo) Run() {
+
+	// Using custom params to decide how much food to eat
+	scaledFloorScore := a.params.currentFloorScore.EvaluateEquation(a.Floor()) / a.params.scalingEquation.EvaluateEquation(a.Floor())
+	scaledHpScore := a.params.currentHpScore.EvaluateEquation(a.HP()) / a.params.scalingEquation.EvaluateEquation(a.HP())
+	foodToEat := math.Max(0.0, 50*scaledFloorScore+50*scaledHpScore)
```
Comment (jzzheng22) [inline pkg/agents/team4/agent1/agent.go:93]:
```suggestion
	foodToEat := food.FoodType(math.Max(0.0, 50*scaledFloorScore+50*scaledHpScore))
```

File: pkg/agents/team4/agent1/agentConfig.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        0.0,
+        0.05739363938985369
+    ],
+    "Hp": [
+        -0.14109909407957166,
+        -0.0782006226777112,
+        -0.1327391844946113,
+        0.0
+    ]
+}
```
Comment (jzzheng22) [inline pkg/agents/team4/agent1/agentConfig.json:14]:
```suggestion
}

```

File: pkg/agents/team4/agent1/bestAgents.json [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            0.05739363938985369
+        ],
+        "Hp": [
+            -0.018037803256623505,
+            -0.12742007202024336,
+            -0.08673948955498505,
+            0.0
+        ]
+    }
+]
```
Comment (jzzheng22) [inline pkg/agents/team4/agent1/bestAgents.json:72]:
```suggestion
]

```

**Review by jzzheng22 — CHANGES_REQUESTED** [review]

File: pkg/agents/team4/agent1/agentConfig.json (hunk 3 above)
Comment (hussain2603) [reply pkg/agents/team4/agent1/agentConfig.json:14]: that's the end bracket for the json file

File: pkg/agents/team4/agent1/bestAgents.json (hunk 4 above)
Comment (hussain2603) [reply pkg/agents/team4/agent1/bestAgents.json:72]: Thats the closing bracket for the file

File: pkg/agents/team4/agent1/agentConfig.json (hunk 3 above)
Comment (jzzheng22) [reply pkg/agents/team4/agent1/agentConfig.json:14]: It's just adding a blank line

File: pkg/agents/team4/agent1/bestAgents.json (hunk 4 above)
Comment (jzzheng22) [reply pkg/agents/team4/agent1/bestAgents.json:72]: It's just adding a blank line

**Review by jzzheng22 — APPROVED** [review]

**Pull request merged**